src/app/account-security/account-password.component.ts

```diff
@@ -0,0 +1,94 @@
+// --------- BEGIN RUNBOX LICENSE ---------
+// Copyright (C) 2016-2018 Runbox Solutions AS (runbox.com).
+//
+// This file is part of Runbox 7.
+//
+// Runbox 7 is free software: You can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// Runbox 7 is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+import { Component } from '@angular/core';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { RMM } from '../rmm';
+import { share, timeout } from 'rxjs/operators';
+
+@Component({
+    selector: 'app-account-password',
+    styleUrls: ['account.security.component.scss'],
+    templateUrl: 'account-password.component.html',
+})
+export class AccountPasswordComponent {
+    old_password: string;
+    new_password: string;
+    confirm_password: string;
+    error: string;
+    is_busy: boolean;
+
+    constructor(public snackBar: MatSnackBar, public rmm: RMM) {
+        this.rmm.me.load();
+    }
+
+    check_password(old_password = this.old_password, new_password = this.new_password, confirm_password = this.confirm_password) {
+        if (old_password && new_password && confirm_password) {
```

Comment:
This would be easier to read (and extend) with a "return early" pattern, like so:

```
if (!old_password || !new_password || !confirm_password) {
    this.error = 'Fields can not be empty';
    return;
}

if (new_password !== confirm_password) {
    this.error = "Passwords don't match";
    return;
}

...

// if we got this far down, the password is good – no need for elses
this.error = undefined;
this.update_password(old_password, new_password, confirm_password);
```